src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -102,18 +102,37 @@ internal static partial class ICodeDefinitionFactoryExtensions
                 // Special case.  If we're generating a struct constructor, then we'll need
                 // to initialize all fields in the struct, not just the ones we're creating.  To
                 // do that, we call the default constructor.
-                var realFields = containingTypeOpt.GetMembers()
```

Comment:
I would rename the containing method entirely.  something like ```bool ShouldGenerateThisConstructorCall(...)```.  And then, the caller uses the result value to generate the empty list, or use 'default'.